Userland/Libraries/LibCrypto/Checksum/IPv4Header.cpp

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2020-2022, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Array.h>
+#include <AK/NumericLimits.h>
+#include <AK/Span.h>
+#include <AK/Types.h>
+#include <LibCrypto/Checksum/IPv4Header.h>
+#include <arpa/inet.h>
+
+namespace Crypto::Checksum {
+
+void IPv4Header::update(ReadonlyBytes data)
+{
+    for (size_t i = 0; i < data.size() / 2; ++i)
+        // Dealing with the byte order isn't technically a part of the checksumming
+        // process, because it's expected that you'd already have the packet decoded,
+        // but since we're dealing with raw data, the responsibility of decoding (also)
+        // falls on us.
+        m_state += ntohs(*reinterpret_cast<u16 const*>(data.data() + (i * 2)));
```

Comment:
2.5 possibly better options:
A: Force Aligned data to be passed in
B: Use a FixedMemoryStream and read BigEndian<u16>
B.5; Reinterpret cast to use BigEndian<u16>[] as that is packed/byte-alligned

Reply:
That also sounds good?